src/Exception/RequestException.php

```diff
@@ -37,19 +40,4 @@ public function getRequest()
     {
         return $this->request;
     }
-
-    /**
-     * @param RequestInterface $request
-     * @param \Exception       $e
-     *
-     * @return RequestException
-     */
-    public static function wrapException(RequestInterface $request, \Exception $e)
```

Comment:
not sure what this was for. i guess we should go for the correct exception and just use new. its also weird how the passed request would get lost if $e is already a RequestException...